chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+
+from chainercv.links.model.faster_rcnn.utils.anchor_target_creator import \
+    AnchorTargetCreator
+from chainercv.links.model.faster_rcnn.utils.proposal_target_creator import \
+    ProposalTargetCreator
+
+
+class FasterRCNNTrainChain(chainer.Chain):
+
+    """Calculate Faster R-CNN losses and report them.
+
+    This is used to train Faster R-CNN in the joint training scheme [#]_.
+
+    .. [#] Shaoqing Ren, Kaiming He, Ross Girshick, Jian Sun. \
+    Faster R-CNN: Towards Real-Time Object Detection with \
+    Region Proposal Networks. NIPS 2015.
+
+    Args:
+        faster_rcnn (~chainercv.links.model.faster_rcnn.FasterRCNN):
+            A Faster R-CNN model to train.
+        rpn_sigma (float): Sigma parameter for localization loss
+            of RPN.
+        sigma (float): Sigma paramter for localization loss of
+            calculated from the output of the head.
+        anchor_target_creator_params (dict): Key valued paramters for
+            :obj:`chainercv.links.model.faster_rcnn.AnchorTargetCreator`.
```

Comment:
Why don't you take an instance of `AnchorTargetCreator` directly? I think it is more simple to pass  faster_rcnn_params` as an argument of `AnchorTargetCreator.__call__`.